jolt-core/src/jolt/vm/bytecode.rs

```diff
@@ -238,8 +233,21 @@ impl<F: PrimeField, G: CurveGroup<ScalarField = F>> BytecodePolynomials<F, G> {
     }
 
     fn preprocess(bytecode: &mut Vec<ELFRow>, trace: &mut Vec<ELFRow>) {
+        for instruction in bytecode.iter_mut() {
+            assert!(instruction.address >= RAM_START_ADDRESS as usize);
+            assert!(instruction.address % BYTES_PER_INSTRUCTION == 0);
+            instruction.address -= RAM_START_ADDRESS as usize;
+            instruction.address /= BYTES_PER_INSTRUCTION;
+        }
+        for instruction in trace.iter_mut() {
+            assert!(instruction.address >= RAM_START_ADDRESS as usize);
+            assert!(instruction.address % BYTES_PER_INSTRUCTION == 0);
+            instruction.address -= RAM_START_ADDRESS as usize;
+            instruction.address /= BYTES_PER_INSTRUCTION;
+        }
+
         // Bytecode: Add single no_op instruction at adddress | ELF + 1 |
-        let no_op_address = bytecode.last().unwrap().address + ADDRESS_INCREMENT;
+        let no_op_address = bytecode.last().unwrap().address + 1;
```

Comment:
We make the no_op_address necessarily expensive to commit to by putting it at bytecode + 1, rather than one of the early register slots. We can handle this later with a smarter leaf hashing / committing algo or make it smaller here.

No need to handle in this PR but could we add a linear issue? See JOLT(CachedHasher) for some context.

Reply:
JOLT-85